aries_cloudagent/messaging/admin/credential_definitions.py

```diff
@@ -129,6 +152,29 @@ class SendCredDefHandler(BaseHandler):
     async def handle(self, context: RequestContext, responder: BaseResponder):
         """Handle received send cred def request."""
         ledger: BaseLedger = await context.inject(BaseLedger)
+        # If no schema record, make one
```

Comment:
building the credential definition attributes from the 'cred_def_json' that is returned from the "create and store cred def" method would eliminate any need to query the ledger, providing a dramatic speed up while reducing network load both locally and on the ledger.